src/ralph/assets/subscribers.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+import logging
+
+import pyhermes
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+
+from ralph.assets.models.assets import Environment, Service, ServiceEnvironment
+from ralph.assets.models.base import BaseObject
+
+logger = logging.getLogger(__name__)
+
+
+def _update_service_owners(service, business_owners, technical_owners):
+    """
+    Update business and technical owners for selected service.
+
+    Args:
+        service: Service
+        business_owners: dict of owners
+        technical_owners: dict of owners
+    """
+    business_owners = [
+        get_user_model().objects.get(username=i['username'])
+        for i in business_owners
+    ]
+    technical_owners = [
+        get_user_model().objects.get(username=i['username'])
+        for i in technical_owners
+    ]
+
+    service.business_owners = business_owners
+    service.technical_owners = technical_owners
+
+
+def _update_service_environments(service, environments):
+    """
+    Add and removes environments for selcted service.
+
+    Args:
+        service: Service
+        environments: list of environment names
+
+    Returns:
+        True if service environment updated
+        False if can not delete ServiceEnvironment
+    """
+    new_envs = []
+    for env_name in environments:
+        new_envs.append(Environment.objects.get_or_create(name=env_name)[0])
+
+    current = set([u.id for u in service.environments.all()])
+    new = set([u.id for u in new_envs])
+    to_delete = current - new
+    to_add = new - current
+    for env_id in to_add:
+        ServiceEnvironment.objects.create(
+            service=service,
+            environment_id=env_id
+        )
+    for env_id in to_delete:
+        service_env = ServiceEnvironment.objects.get(
+            service=service, environment_id=env_id
+        )
+        if BaseObject.objects.filter(service_env=service_env).exists():
+            logger.error(
+                'Can not delete service environment because when assigned to objects.',  # noqa
+                extra={
+                    'service_uid': service.uid,
+                    'service_env': service_env
+                }
+            )
+            return False
+        service_env.delete()
+
+    return True
+
+
+@pyhermes.subscriber(topic=settings.HERMES_SERVICE_TOPICS['CREATE'])
+@pyhermes.subscriber(topic=settings.HERMES_SERVICE_TOPICS['UPDATE'])
+@pyhermes.subscriber(topic=settings.HERMES_SERVICE_TOPICS['REFRESH'])
+def update_service_handler(service_data):
+    """
+    Add, remove and update Service from hermes event.
```

Comment:
Hmm... Looking at the decorators here, I see "create", "update" and "refresh" - I guess that "remove" is included in "refresh" then..? If so, then maybe add some explaination here..?